app/services/invoices/payments/stripe_service.rb

```diff
@@ -155,6 +157,21 @@ def update_invoice_status(status)
         invoice.update!(status: status)
       end
 
+      def update_wallet(status)
+        return unless invoice.invoice_type == 'credit'
+        return unless status == 'succeeded'
+
+        wallet_transaction = invoice.fees.find_by(fee_type: 'credit')&.invoiceable
+
+        return unless wallet_transaction
+        return if wallet_transaction.status == 'settled'
+
+        wallet_transaction.update!(status: :settled, settled_at: Time.zone.now)
```

Comment:
can we do this a in a separated service? the invoice stripe service should keep his own scope and does not update other objects than invoices

Reply:
Refactored